synapse/rest/client/v1/register.py

```diff
@@ -284,9 +284,9 @@ def _do_password(self, request, register_json, session):
                 400, "Cannot change user ID during registration"
             )
 
-        password = register_json["password"].encode("utf-8")
+        password = register_json["password"]
```

Comment:
uhhh, this looks correct, but also like it's going to break something. any ideas wtf is going on in the existing code here?